### PR TITLE
Task7: Create endpoint postCommentForArticle

### PR DESCRIPTION
**Pull request by ccdu1010**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -144,10 +144,144 @@ describe("app", () => {
...
+  test("400: adding a valid comment for an article that does not exist responds with status 400", () => {
+    const testArticleId = 4000;
+    const requestBody = {
+      username: 'icellusedkars',
+      body: 'Lobster pot'
+    };
+    return request(app)
+     .post(`/api/articles/${testArticleId}/comments`)
+     .send(requestBody)
+     .expect(400);
```
Comment (aloush-dev) [inline __tests__/app.test.js]: This case should really return a 404 not found. as it's not a bad request.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -144,10 +144,144 @@ describe("app", () => {
...
+  test("400: adding a valid comment for a user that does not exist responds with status 400", () => {
+    const testArticleId = 4;
+    const requestBody = {
+      username: 'notrealuser',
+      body: 'Lobster pot'
+    };
+    return request(app)
+     .post(`/api/articles/${testArticleId}/comments`)
+     .send(requestBody)
+     .expect(400);
```
Comment (aloush-dev) [inline __tests__/app.test.js]: Same with this one its a 404 it's a valid request but the username was not found.

File: models/articles.model.js [hunk 3]
```diff
@@ -49,4 +50,60 @@ exports.selectAllArticles = () => {
...
+}
+
+exports.insertCommentForArticle = (articleId, newComment) => {
+  const parsedArticleId = parseInt(articleId);
+  if(isNaN(parsedArticleId)) {
+    return Promise.reject({
+      status: 400,
+      msg: `The article_id '${articleId}' is not a valid number`,
+    });
+  }
```
Comment (itsaliabdallah) [inline models/articles.model.js:86]: isNan is not very reliable as a method to check this. PSQL should throw an error if the ID you've given is not a number.

**Review by aloush-dev — CHANGES_REQUESTED** [review]
I'd like to see another test added for a 201, that if there is extra properties on the object which you send your server will ignore them and post the comment.

Your testing is ok, but for a lot of your eror testing you're only testing the status code in which you are getting back, I'd like to see you testing for some sort of error message you're getting back.

Please implement changes and resubmit, Thanks.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -156,13 +162,19 @@ describe("app", () => {
     const testArticleId = 40000;
     return request(app)
     .get(`/api/articles/${testArticleId}/comments`)
-    .expect(404);
+    .expect(404)
+    .then(({body}) => {
+      expect(body.msg).toBe("No comments found for article_id: 40000")
```
Comment (dappernerddesigns) [inline __tests__/app.test.js]: This should be relating to the article not existing, rather than the comments.

File: errors/index.js [hunk 5]
```diff
@@ -5,8 +5,9 @@ exports.handleCustomErrors = (err, req, res, next) => {
 };
 
 exports.handlePsqlErrors = (err, req, res, next) => {
+    const invalid_text_representation = '22P02';
     const foreign_key_violation = '23503';
-    if (err.code === foreign_key_violation) {
+    if (err.code === foreign_key_violation || err.code === invalid_text_representation) {
```
Comment (dappernerddesigns) [inline errors/index.js:10]: I really like this in your error handler, we can hope people know what the codes mean but these are really good variable names.

**Review by dappernerddesigns — APPROVED** [review]
Good work CC, well done. Merge away!

**Pull request merged**